Integrations/src/main/java/io/deephaven/integrations/python/PyLogOutputStream.java

```diff
@@ -0,0 +1,47 @@
+package io.deephaven.integrations.python;
+
+import org.jetbrains.annotations.NotNull;
+import org.jpy.PyObject;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.function.Supplier;
+
+/**
+ * Simple output stream that redirects all writes to a python io.TextIOBase type.
+ */
+public class PyLogOutputStream extends OutputStream {
+    private final Supplier<PyObject> rawIoBaseSupplier;
+
+    public PyLogOutputStream(Supplier<PyObject> rawIoBaseSupplier) {
+        this.rawIoBaseSupplier = rawIoBaseSupplier;
+    }
+
+    @Override
+    public void write(int i) throws IOException {
+        write(new byte[] {(byte) i});
+    }
+
+    @Override
+    public void write(@NotNull byte[] b) throws IOException {
+        // noinspection PrimitiveArrayArgumentToVarargsMethod
+        rawIoBaseSupplier.get().callMethod("write", new String(b));
```

Comment:
Do we need / want to account for Charset here?

Reply:
IMO, it would be better if we could call `org.jpy.PyObject#call(java.lang.Class<T>, java.lang.String, java.lang.Class<?>[], java.lang.Object[])` - this allows us to be more explicit w/ the parameter and return types.

As it stands now, all of the calls to `callMethod` return `PyObject`, which I think is extra garbage we shouldn't need to create.

Reply:
Yeah, that raises a question - why can jpy convert cleanly from py bytes to java byte[], but not vice versa? That's the real answer here, since it is already encoded (and we can't cleanly ask for the underlying encoding... since jpy might break it).

Reply:
That said, why be more explicit with the param types? Both call into `PyLib.callAndReturnValue`, but the variant that takes lots of `Class<?>`s does java-side type checks, rather than just passing the string instance - and there is no return type to be concerned with.

(Also, `.call()` vs `.callMethod()` has semantics that _technically_ we don't want here, we actually want `callMethod`. However, as far as I can tell, jpy just throws away that extra flag with a "todo why isn't this used?" sort of note)

Reply:
To be clear, these are not the same - `call` sets methodCall=false and its docs clarify that this is invoking a python callable object, not a method. All overrides do this.

In contrast, `callMethod` sets methodCall=true, and is intended for calling methods. We don't have (but arguably should?) an overload of `callMethod` like `call` does to support each arg type. I'd support going so far as to add overloads that return primitives or void, to avoid that extra cost you describe.

In this case, we want to call a method, we are not interacting with a callable object.

Reply:
the jpy implementation does not actually use that boolean internally - some sort of historic wart before my time.

Reply:
Yes - that's what I said above - in which case we should either figure out what issue might be there or historically have been there, or change the API - I'm just going off of the current stated contract of JPy, which is that this isn't to be used for method calls. But at the end of the day, we're using the wrong method that happens to do the right thing, for now, probably, to save on creating one object (that will either be called approx once per jvm, in the case of close(), or for write and maybe flush, are likely to be jitted to be inlined and have escap analysis handle anyway)?

I've you insist, I'll write the wrong code that currently does what we want but not what we're asking for, but this is an improvement to make to jpy and then improve our usage of it, so I'm objecting to deliberately doing the wrong thing.

Reply:
I've filed https://github.com/jpy-consortium/jpy/issues/87.

`org.jpy.PyObject#call(java.lang.Class<T>, java.lang.String, java.lang.Class<A0>, A0)` doesn't expose the notion of "methodCall", or have any javadoc (my fault). I think your argument falls down a little bit though, b/c if the implementation of that method instead used `true` instead of `false`, the outcome would be exactly the same, and is arguably an implementation detail.

I do like to consider myself against unnecessary pre-optimization. With respect to PyObject though, it's not the allocation itself I'm usually worried about, but the implications that extraneous PyObjects may cause (PyObjectReferences growth, unnecessarily keeping python references alive). And specifically as a glue layer between streams, there's potential for this to be heavily used. In general, if there is a way to accomplish the same outcome but without creating a PyObject, that's the direction I'll lean.

All that said - I've argued my case, and can let the code stand as-is if that's still your preference.